### PR TITLE
Problems API documentation improvements

### PR DESCRIPTION
**Pull request by donat**
And add backlink from sample to documentation.

Fixes https://github.com/gradle/gradle-private/issues/4562

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by donat** [comment]
@bot-gradle test this

File: platforms/documentation/docs/src/docs/userguide/authoring-builds/plugins/reporting_problems.adoc
Comment (hegyibalint) [inline platforms/documentation/docs/src/docs/userguide/authoring-builds/plugins/reporting_problems.adoc:1]: ⚠️ Even if out of scope of this PR, I realized that this document is out of date

Comment (hegyibalint) [reply platforms/documentation/docs/src/docs/userguide/authoring-builds/plugins/reporting_problems.adoc:1]: (we still refer namespaces in the docs)

**Review by lkasso — APPROVED** [review]
This is amazing. A great addition to the User Manual.

**Review by reinsch82 — APPROVED** [review]
LGTM

**Comment by donat** [comment]
@bot-gradle test RFM

**Comment by donat** [comment]
@bot-gradle merge

**Pull request merged**